### DISCUSSION TITLE
Issues with Redash and Atlas MongoDB ver 7

### DISCUSSION THREAD

**Discussion by JohnDimopanagos** [getredash/redash#7045] [Q&A]
Hi there,

I am using Redash AWS AMI and upgraded to version 10. 
Trying to use MongoDB v7 but without any luck.
The datasource is being created and test connection success, but when I am trying to make a query I get no collections.
Also I see that on console i get this error:

Refused to load blob:https://mysitedomain/1ec42eec-62bc-4533-b490-6460cec39306 because it does not appear in the worker-src directive of the Content Security Policy.

I dont know if it helps.

**Comment by h-sasakawa** [comment]
@JohnDimopanagos I think the current tagged version of Redash using this version of pymongo client https://pypi.org/project/pymongo/4.3.3/ , but the client doesn't support mongo v7.
This PR created by dependabot upgraded pymongo to newer version supports mongo v7, but the changes haven't released yet.
https://github.com/getredash/redash/pull/6863

It might work if you build the main branch.

**Reply by justinclift** [reply]
That update is also in the `preview` Redash image on Docker Hub:

https://hub.docker.com/r/redash/redash/tags?name=preview

To use that, you'll need to make a backup of your existing database (*DO NOT* skip this), and upgrade the database using the instructions here:

https://hub.docker.com/r/redash/redash/tags?name=preview

**Reply by ej2015** [reply]
I'm facing similar issue with self-hosted MongoDB v7.
redash: https://artifacthub.io/packages/helm/redash/redash
chart version: 3.1.0-alpha-9
redash image tag: docker.io/redash/preview:24.04.0-dev-b8718551048.32
pymongo: pymongo = { version = "4.6.3", extras = ["srv", "tls"] }

mongodb: https://artifacthub.io/packages/helm/bitnami/mongodb
chart version: mongodb-15.6.0 
mongo: docker.io/bitnami/mongodb:7.0.9-debian-12-r4

connection string in redash: mongodb://user:pw@mongodb-headless.mongo:27017/?authSource=admin&replicaSet=rs0
and i set username and password in the respective fields

keep getting this error:
`Authentication failed., full error: {'ok': 0.0, 'errmsg': 'Authentication failed.', 'code': 18, 'codeName': 'AuthenticationFailed', '$clusterTime': {'clusterTime': Timestamp(1729128455, 1), 'signature': {'hash': b'\x1b\x13E\xff\x99\t%\xc2\xf4(yf\xd3\xcf\x89\xf1\xf0\xca\x84\x1f', 'keyId': 7420962562679767042}}, 'operationTime': Timestamp(1729128455, 1)}`

btw my other application using java client is able to connect to the mongodb 

anyone can help? thx

**Reply by justinclift** [reply]
Looking through the `pymongo` documentation, support for MongoDB 7.0 was added in version 4.4 of `pymongo`: https://pymongo.readthedocs.io/en/stable/changelog.html#changes-in-version-4-4

As you point out, we're using a newer `pymongo` than that (`4.6.3`), so whatever the problem is it's not a case of pymongo being too old.

On the "version of pymongo" topic, it looks like newer 4.x releases of PyMongo have breaking changes, so will need some extra care and checking when we get around to updating it: https://pymongo.readthedocs.io/en/stable/changelog.html#id7

---

The pymongo docs also have a stub page with info on connecting to Atlas:

https://pymongo.readthedocs.io/en/stable/atlas.html

We *might* need to do some investigation there to ensure we're correctly setting things so that Atlas works.

@ej2015 The connection string you have there doesn't seem to have a `tls` parameter:

```
mongodb://user:pw@mongodb-headless.mongo:27017/?authSource=admin&replicaSet=rs0
```

The docs there seem to indicate there are two ways of potentially setting TLS, so maybe we're doing it "the other way".  If we're not though, and there's really a lack of TLS, then that's probably the thing which needs changing.
